src/memo/memo.ml

```diff
@@ -204,25 +204,19 @@ let global_dep_dag = Dag.create ()
 
 (* fiber context variable keys *)
 let call_stack_key = Fiber.Var.create ()
-let get_call_stack =
-  Fiber.Var.get call_stack_key (* get call stack *)
-  >>| Option.value ~default:[] (* default call stack is empty *)
+let get_call_stack () = Fiber.Var.get call_stack_key |> Option.value ~default:[]
 
 let push_stack_frame frame f =
-  get_call_stack >>= fun stack ->
+  let stack = get_call_stack () in
   Fiber.Var.set call_stack_key (frame :: stack) f
 
-let dump_stack v =
-  get_call_stack
-  >>|
-  (Printf.printf "Memoized function stack:\n";
-   List.iter ~f:(
-     fun st -> Printf.printf "   %s %s\n"
-                 (Stack_frame.name st)
-                 (Stack_frame.input st |> Sexp.to_string)
-   )
-  )
-  >>| (fun _ -> v)
+let dump_stack () =
+  let stack = get_call_stack () in
+  Printf.printf "Memoized function stack:\n";
+  List.iter stack ~f:(
+    fun st -> Printf.printf "   %s %s\n"
```

Comment:
I just noticed this, but we should probably print this to stderr. This is how we log everything as well.

Reply:
Good call

Reply:
Fixed